src/context/trace_context.rs

```diff
@@ -259,6 +264,7 @@ impl TracingContext {
 
         let mut span = Box::new(Span::new(
             self.next_span_id,
+            -1,
```

Comment:
Why hard code the span parent Span ID? Parent span ID should be the previous unfinished span in this tracing context.

Reply:
If `/status/500` fails, and retry `/status/400` again, then yes, the right is correct.
But this example seems not relative to why hardcode parent span ID? The right logic is, `parent span ID` should be the latest unfinished span ID in this context/thread.
I don't mean the original codes are correct, I don't check so carefully. But the new one seems not correct at least.

Reply:
I think I understand what you mean. In this case, the parent span ID should be set to the Span ID of the previous span that has not set end time. The code will change a lot.

Reply:
I think you just need to add a current span ref in `TracingContext`, and always use it as parent span ID, unless a parent Span ref is given as `new span`'s parameter.
The current span in `TracingContext` should be set to its parent when it is closed.

In the Java agent, we use a stack<Span>(implement through a LinkedList) to push/pop the active span. 
You could take this as a good reference, 

https://github.com/apache/skywalking-java/blob/372d75403053c1eb81d3cc4ec696a675fab1da71/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java#L80

Reply:
I can see you did a lot of work on making rust-sdk in PHP ecosystem, and also helped our php-agent in skyapm org.
If you are willing to take more responsibility to this repo, this would be a very good start.

@Shikugawa implemented this for the first time, we have a lot to do to polish.